samples/node/shadow/index.ts

```diff
@@ -0,0 +1,355 @@
+                        else {
+                            console.log("Desired value not found in delta. Skipping..");
+                        }
+                    }
+                }
+                else {
+                    console.log("Delta did not report a change in '" + shadow_property + "'.");
+                }
+
+                resolve(true);
+            }
+
+            console.log("Subscribing to Delta events..");
+            const deltaShadowSubRequest: iotshadow.model.ShadowDeltaUpdatedSubscriptionRequest = {
+                thingName: argv.thing_name
+            };
+
+            await shadow.subscribeToShadowDeltaUpdatedEvents(
+                deltaShadowSubRequest,
+                mqtt.QoS.AtLeastOnce,
+                (error, response) => deltaEvent(error, response));
+
+            resolve(true);
+        }
+        catch (error) {
+            reject(error);
+        }
+    });
+}
+
+async function get_current_shadow(shadow: iotshadow.IotShadowClient, argv: Args) {
+    return new Promise(async (resolve, reject) => {
+        try {
+            const getShadow: iotshadow.model.GetShadowRequest = {
+                thingName: argv.thing_name
+            }
+
+            shadow_update_complete = false;
+            console.log("Requesting current shadow state..");
+            shadow.publishGetShadow(
+                getShadow, 
+                mqtt.QoS.AtLeastOnce);
+            
+            await get_current_shadow_update_wait();
+            resolve(true);
+        }
+        catch (error) {
+            reject(error);
+        }
+    });
+}
+
+
+async function get_current_shadow_update_wait() {
+    // Wait until shadow_update_complete is true, showing the result returned
+    return await new Promise(resolve => {
+      const interval = setInterval(() => {
+        if (shadow_update_complete==true) {
+          resolve(true);
+          clearInterval(interval);
+        };
+      }, 200);
+    });
+  }
+
+function change_shadow_value(shadow: iotshadow.IotShadowClient, argv: Args, new_value?: object) { 
+    return new Promise(async (resolve, reject) => {
+        try {
+            if (typeof new_value !== 'undefined') {
+                let new_value_any : any = new_value;
+                let skip_send = false;
+
+                if (new_value_any !== null) {
+                    if (new_value_any[shadow_property] == shadow_value) {
+                        skip_send = true;
+                    }
+                }
+                if (skip_send == false) {
+                    if (new_value_any === null) {
+                        shadow_value = new_value_any;
+                    }
+                    else {
+                        shadow_value = new_value_any[shadow_property];
+                    }
+
+                    console.log("Changed local shadow value to '" + shadow_value + "'.");
+                    console.log("Update reported shadow value to '" + shadow_value + "'..")
```

Comment:
You don't have to keep both of these. You can remove the "Update reported..." because below you have a similar log message